arangod/Aql/QuerySnippet.cpp

```diff
@@ -225,8 +224,12 @@ void QuerySnippet::serializeIntoBuilder(
           TRI_ASSERT(colAcc != nullptr);
           if (colAcc->collection() == distCollection) {
             // Found one, use all shards of it
-            for (auto const& s : exp.second) {
-              distIds.emplace_back(s);
+            // unordered_map of collections to sets of ids
+            for (auto const& collectionMap : exp.second) {
+              // set of shard ids
```

Comment:
Please assert that collectionMap only has 1 entry in this case.
And that this entry is equal to colAcc->collection()